src/query/storage/m3/consolidators/multi_fetch_result.go

```diff
@@ -223,33 +247,51 @@ func (r *multiResult) Add(
 			r.dedupeMap = newTagDedupeMap(opts)
 		}
 
-		r.addOrUpdateDedupeMap(r.seenFirstAttrs, first)
-		return
+		added = r.addOrUpdateDedupeMap(r.seenFirstAttrs, first)
+	} else {
+		// Now de-duplicate
+		added = r.addOrUpdateDedupeMap(attrs, newIterators)
 	}
 
-	// Now de-duplicate
-	r.addOrUpdateDedupeMap(attrs, newIterators)
+	// the series limit was reached by adding the results of this namespace to the existing results.
```

Comment:
this is the case where the query series limit is exhausted across multiple namespaces.

Reply:
unfortunately since results from one namespace can overwrite results from another namespace, we have to iterate through the results to see if the limit was exceeded. ie. can't rely on metadata.FetchSeriesCount

Reply:
good catch. probably need to guard this with `r.err == nil` as well. not a big fan of apis that don't return errors